aiida/cmdline/params/types/path.py

```diff
@@ -31,3 +33,76 @@ def convert(self, value, param, ctx):
 
     def __repr__(self):
         return 'ABSOLUTEPATH'
+
+
+class ImportPath(click.Path):
+    """AiiDA extension of Click's Path-type to include URLs
+    An ImportPath can either be a `click.Path`-type or a URL.
+    """
+
+    # pylint: disable=protected-access
+
+    def __init__(self, timeout_seconds=None, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+        self.timeout_seconds = timeout_seconds
+
+    def convert(self, value, param, ctx):
+        """Overwrite `convert`
+        Check first if `click.Path`-type, then check if URL.
+        """
+        try:
+            # Check if `click.Path`-type
+            return super(ImportPath, self).convert(value, param, ctx)
+        except click.exceptions.BadParameter:
+            # Check if URL
+            return self.checks_url(value, param, ctx)
+
+    def checks_url(self, value, param, ctx):
+        """Do checks for possible URL path"""
+        from six.moves import urllib
+        from django.core.validators import URLValidator
+        from django.core.exceptions import ValidationError as URLValidationError
+        from socket import timeout
+
+        url = value
+
+        try:
+            URLValidator()(url)
+        except URLValidationError:
+            self.fail('"%s" is neither a valid %s nor a valid URL.' % (click._compat.filename_to_ui(url), self.name),
+                      param, ctx)
+
+        try:
+            urllib.request.urlopen(url, data=None, timeout=self.timeout_seconds)
+        except (urllib.error.URLError, urllib.error.HTTPError, timeout):
+            self.fail(
+                '%s "%s" could not be reached within %d s.' % (self.path_type, click._compat.filename_to_ui(url),
+                                                               self.timeout_seconds), param, ctx)
+
+        return url
+
+    @property
+    def timeout_seconds(self):
+        return self._timeout_seconds
+
+    # pylint: disable=attribute-defined-outside-init
+    @timeout_seconds.setter
+    def timeout_seconds(self, value):
+        if isinstance(value, str):
```

Comment:
This whole block can be cleaned up. First off, you can get either a `ValueError` or `TypeError` if the type of the timeout is incorrect. Furthermore, the combination of `if` conditional and then try-except is superfluous and not very pythonic. I would change this whole block simply in:
```
try:
     self._timeout_seconds = int(value)
except ValueError:
    raise TypeError('timeout should be an integer but got: {}'.format(type(value)))
```
The default should simply be set in the constructor of this object.

Reply:
Done. I have reduced it to your more simple try-except block, keeping the following limit checks.
The default value is set in the constructor with a check `if timeout_seconds is None` set it to the default.
I think it is maybe better to keep a None value in the `__init__` parameters, since if the default is set already here and someone should pass a `None` value the default value will still be used. Unless we specifically want it to raise if a None value is passed?